PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfigLoader.java

```diff
@@ -157,11 +157,12 @@ private RuleConfig getRuleConfig(JsonReader reader) throws IOException {
                 case "servicesToCheck":
                     servicesToCheck = getListFromJsonArray(reader);
                     break;
-                case "regexPatterns":
-                    listedItemsToCheck = getValuesFromJsonReader(reader);
-                    break;
                 default:
-                    if (fieldName.endsWith("Check")) {
+                    if (fieldName == "typesToCheck") {
+                        listedItemsToCheck = getListFromJsonArray(reader);
+                    } else if (fieldName == "regexPatterns") {
+                        listedItemsToCheck = getValuesFromJsonReader(reader);
```

Comment:
I remember having an issue with switch cases at this stage because of the duplicate `listedItemsToCheck` but I think that was resolved elsewhere because bringing back the switch cases now is not problematic anymore